filter/filter.go

```diff
@@ -606,6 +610,71 @@ func (c *Client) UpdateBlueprint(ctx context.Context, userAuthToken, serviceAuth
 	return m, eTag, nil
 }
 
+// SubmitFilter function to submit the request to submit a filter for a cantabular dataset.
+// Should POST to /filters/{filterid}/submit in dp-cantabular-filter-flex-api microservice.
+func (c *Client) SubmitFilter(ctx context.Context, userAuthToken, serviceAuthToken, downloadServiceToken, ifMatch string, sfr SubmitFilterRequest) (*SubmitFilterResponse, string, error) {
+	b, err := json.Marshal(sfr)
+	if err != nil {
+		return nil, "", errors.Wrap(err, "could not unmarshal")
+	}
+
+	uri := fmt.Sprintf("%s/filters/%s/submit", c.hcCli.URL, sfr.FilterID)
+
+	clientlog.Do(ctx, "updating filter job", service, uri, log.Data{
+		"method": postMethod,
+		"body":   string(b),
+	})
+
+	req, err := http.NewRequest(postMethod, uri, bytes.NewBuffer(b))
+	if err != nil {
+		return nil, "", errors.Wrap(err, "could not create a new POST request")
+	}
+
+	if err = headers.SetAuthToken(req, userAuthToken); err != nil {
+		return nil, "", errors.Wrap(err, "failed to set auth token")
+	}
+	if err = headers.SetServiceAuthToken(req, serviceAuthToken); err != nil {
+		return nil, "", errors.Wrap(err, "failed to set service auth token")
+	}
+	if err = headers.SetDownloadServiceToken(req, downloadServiceToken); err != nil {
+		return nil, "", errors.Wrap(err, "failed to set download service token")
+	}
+	if err = headers.SetIfMatch(req, ifMatch); err != nil {
+		return nil, "", errors.Wrap(err, "failed to set if match")
+	}
+
+	resp, err := c.hcCli.Client.Do(ctx, req)
+	if err != nil {
+		return nil, "", errors.Wrap(err, "failed to create a new http client")
+	}
+	defer closeResponseBody(ctx, resp)
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
Would it be better to move this check till after we have the response body, and then return the response in the error logData?